app/models/translation_override.rb

```diff
@@ -129,6 +129,13 @@ def self.custom_interpolation_keys(translation_key)
   private_class_method :i18n_changed
   private_class_method :expire_cache
 
+  def original_translation_deleted?
+    I18n.overrides_disabled { I18n.t!(transformed_key, locale: :en) }
+    false
```

Comment:
I think it would be better to use this instead:

```suggestion
    I18n.overrides_disabled { I18n.t!(transformed_key, locale: :en) }.is_a?(String)
```

Let's assume someone created a translation override when `js.number.human.storage_units.units.byte` wasn't a pluralized string with one/other yet. A couple of weeks later we notice our mistake, update the English locale file and make it pluralized as it is now. Your current method would still detect it as "not deleted" even though the `js.number.human.storage_units.units.byte` has been deleted and was replaced by `js.number.human.storage_units.units.byte.one` and `js.number.human.storage_units.units.byte.other`.

And this is not a theoretical case -- this happens every couple of months.

Unless I'm missing something, your current code doesn't handle that case. Checking if the existing translation in the English locale file is actually a string should solve that problem. Can you add a spec for that case as well?

Reply:
Nice catch! Didn't know that if there's a nested key `I18n.t` will pass you an entire hash. 😅